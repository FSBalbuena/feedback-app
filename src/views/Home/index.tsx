import Grid from "@mui/material/Grid2";
import Button from "@mui/material/Button";
import Typography from "@mui/material/Typography";
import useIsMobile from "@/hooks/useIsMobile";
import { Home as copyText } from "@/data/copy";

type Props = {
  onStart: () => void;
};
function Home({ onStart }: Props) {
  const isMobile = useIsMobile();
  return (
    <Grid
      container
      height={"100vh"}
      width={"100vw"}
      justifyContent={"center"}
      alignItems={"center"}
    >
      <Grid
        container
        maxWidth="md"
        spacing={4}
        direction={"column"}
        size={{ xs: 10, md: 8 }}
      >
        <Grid textAlign={"center"}>
          <Typography variant={isMobile ? "h5" : "h3"} gutterBottom>
            {copyText.TITLE}
          </Typography>
          {isMobile ? null : (
            <Typography component={"p"} variant="h5" gutterBottom>
              {copyText.DESCRIPTION}
            </Typography>
          )}
        </Grid>
        <Grid textAlign={"center"}>
          <Button variant="contained" size="large" onClick={onStart}>
            {copyText.START_BUTTON}
          </Button>
        </Grid>
      </Grid>
    </Grid>
  );
}

export default Home;
