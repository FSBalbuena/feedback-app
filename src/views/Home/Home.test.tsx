import Home from "./index";
import { render } from "@testing-library/react";

describe("Home", () => {
  it("should render", () => {
    const { container } = render(<Home onStart={() => {}} />);
    expect(container).toMatchSnapshot();
  });
});
