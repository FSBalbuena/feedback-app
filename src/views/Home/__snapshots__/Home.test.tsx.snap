// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render 1`] = `
<div>
  <div
    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-smhp44-MuiGrid2-root"
  >
    <div
      class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-column MuiGrid2-grid-xs-10 MuiGrid2-grid-md-8 MuiGrid2-spacing-xs-4 css-7pw7xi-MuiGrid2-root"
    >
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row css-1mgi31c-MuiGrid2-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-uynfxu-MuiTypography-root"
        >
          Confidently give feedback that empowers others and strengthens relationships.
        </h3>
        <p
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1f9xtph-MuiTypography-root"
        >
          If you’re feeling unsure, anxious, or stuck about delivering constructive feedback, this is the place to find clarity, tools, and support. Together, we’ll turn your fears into confidence and your feedback into a positive force for growth.
        </p>
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row css-1mgi31c-MuiGrid2-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-79zo2g-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Get Started
        </button>
      </div>
    </div>
  </div>
</div>
`;
